File: CourseMembers/DataProvider.m
//
//  DataProvider.m
//  CourseMembers
//
//  Created by Jonas Hansen on 16/09/15.
//  Copyright (c) 2015 Jonas Hansen. All rights reserved.
//

#import "DataProvider.h"
#import "PersonData.h"

@implementation DataProvider

static DataProvider* dataProvider;

+(DataProvider*) sharedDataProvider{
    if(dataProvider == nil){
        dataProvider = [[DataProvider alloc]init];
    }
    return dataProvider;
}

-(instancetype)init {
    // Names
    NSArray* memberNames = [NSArray arrayWithObjects:
                            @"Jonas",
                            @"Michi",
                            @"Silvio",
                            @"Steffi",
                            @"Dorian",
                            @"Rainer",
                            nil
    ];
    self.memberNames = memberNames;
    
    // Persons
    NSArray* memberPersons = [NSArray arrayWithObjects:
                              [[PersonData alloc] initWithFirstName:@"Jonas" lastName:@"Hansen" andPLZ:6218],
                              [[PersonData alloc] initWithFirstName:@"Michael" lastName:@"Zurmühle" andPLZ:1],
                              [[PersonData alloc] initWithFirstName:@"Silvio" lastName:@"Stappung" andPLZ:1337],
                              [[PersonData alloc] initWithFirstName:@"Steffanie" lastName:@"Vogel" andPLZ:2],
                              [[PersonData alloc] initWithFirstName:@"Dorian" lastName:@"der Übermensch" andPLZ:-1],
                              [[PersonData alloc] initWithFirstName:@"Rainer" lastName:@"Winkler" andPLZ:91448],
                              nil
    ];
    self.memberPersons = memberPersons;
    
    return self;
}

@end

File: SW02-SwitfProgrammers/SwiftProgrammers/SwiftProgrammers/DataProvider.swift
//
//  DataProvider.swift
//  SwiftProgrammers
//
//  Created by Jonas Hansen on 23/09/15.
//  Copyright © 2015 Jonas Hansen. All rights reserved.
//

import Foundation

class DataProvider{
    static var sharedInstance : DataProvider = DataProvider()
    var memberNames : [String]
    var memberPersons : [Person]
    
    private init(){
        memberNames = ["Jonas","Michi","Stefi","Silvio"]
        
        memberPersons = [
            Person(firstname: "Jonas", lastname: "Hansen", plz: 6218),
            Person(firstname: "Michi", lastname: "Zurmühle", plz: 1),
            Person(firstname: "Stefanie", lastname: "Vogel", plz: 2),
            Person(firstname: "Silvio", lastname: "Stappung", plz: 1337),
            Person(firstname: "Rainer", lastname: "Winkler", plz: 91448)
        ]
    }
}